prisma/schema.prisma

```diff
@@ -510,3 +511,57 @@ model PaymentSourceConfig {
   rpcProvider       RPCProvider
   PaymentSource     PaymentSource?
 }
+
+// Webhook models
+model WebhookEndpoint {
+  id        String   @id @default(cuid())
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  url       String
+  authToken String 
+  events    String[] 
+  name      String? 
+  
+  PaymentSource   PaymentSource? @relation(fields: [paymentSourceId], references: [id], onDelete: Cascade)
+  paymentSourceId String?
+
+  isActive             Boolean   @default(true)
+  failureCount         Int       @default(0)
+  consecutiveFailures  Int       @default(0)
+  lastSuccessAt        DateTime?
+  disabledAt           DateTime?
+
+  WebhookDeliveries WebhookDelivery[]
+}
+
+model WebhookDelivery {
+  id        String   @id @default(cuid())
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+
+  WebhookEndpoint   WebhookEndpoint @relation(fields: [webhookEndpointId], references: [id], onDelete: Cascade)
+  webhookEndpointId String
+
+  eventType String 
```

Comment:
I think we need to have this typesafe. basically add event types as a enum and we provide docs how the data/payload will look. It has to be a predefined type though.